hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/services/ITestReadBufferManager.java

```diff
@@ -0,0 +1,119 @@
+package org.apache.hadoop.fs.azurebfs.services;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FSDataInputStream;
+import org.apache.hadoop.fs.FSDataOutputStream;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.azurebfs.AbstractAbfsIntegrationTest;
+import org.apache.hadoop.fs.azurebfs.AzureBlobFileSystem;
+import org.assertj.core.api.Assertions;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.*;
+
+import static org.apache.hadoop.fs.azurebfs.constants.ConfigurationKeys.*;
+import static org.apache.hadoop.fs.azurebfs.constants.FileSystemConfigurations.*;
+
+public class ITestReadBufferManager extends AbstractAbfsIntegrationTest {
+
+    public ITestReadBufferManager() throws Exception {
+    }
+
+    @Test
+    public void testPurgeBufferManagerForParallelStreams() throws Exception {
+        describe("Testing purging of buffers from ReadBufferManager for " +
+                "parallel input streams");
+        final int numBuffers = 16;
+        final LinkedList<Integer> freeList = new LinkedList<>();
+        for (int i=0; i < numBuffers; i++) {
+            freeList.add(i);
+        }
+        AzureBlobFileSystem fs = getABFSWithReadAheadConfig();
+        for(int i=0; i < 4; i++) {
+            String fileName = methodName.getMethodName() + i;
+            byte[] fileContent = getRandomBytesArray(ONE_MB);
+            Path testFilePath = createFileWithContent(fs, fileName, fileContent);
+            try (FSDataInputStream iStream = fs.open(testFilePath)) {
+                iStream.read();
+            }
+        }
+        ReadBufferManager bufferManager = ReadBufferManager.getBufferManager();
+        Assertions.assertThat(bufferManager.getCompletedReadListCopy().size())
+                .describedAs("After closing all streams completed list size should be 0")
+                .isEqualTo(0);
+
+        Assertions.assertThat(bufferManager.getInProgressCopiedList().size())
+                .describedAs("After closing all streams inProgress list size should be 0")
+                .isEqualTo(0);
```

Comment:
assertThat(list).hasSize(), as you do below.
could consider an AsserListEmpty(text, list) with that operation